File: Integraupt-Backend/BackendUsuarios/src/main/java/Controller/DocenteController.java
package controller;

import dto.DocenteForm;
import dto.EstadoUsuarioForm;
import java.util.List;
import model.Docente;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import service.docente.DocenteService;

@RestController
@RequestMapping("/api/docentes")
public class DocenteController {

    private final DocenteService service;

    public DocenteController(DocenteService service) {
        this.service = service;
    }

    @GetMapping
    public List<Docente> listar() {
        return service.listar();
    }

    @GetMapping("/{id}")
    public ResponseEntity<Docente> obtener(@PathVariable Integer id) {
        try {
            Docente docente = service.obtener(id);
            return ResponseEntity.ok(docente);
        } catch (RuntimeException e) {
            return ResponseEntity.notFound().build();
        }
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public Docente crear(@RequestBody DocenteForm form) {
        return service.registrar(form);
    }

    @PutMapping("/{id}")
    public ResponseEntity<Docente> actualizar(@PathVariable Integer id, @RequestBody DocenteForm form) {
        try {
            Docente docente = service.actualizar(id, form);
            return ResponseEntity.ok(docente);
        } catch (RuntimeException e) {
            return ResponseEntity.notFound().build();
        }
    }

    @PatchMapping("/{id}/estado")
    public ResponseEntity<Docente> actualizarEstado(@PathVariable Integer id, @RequestBody EstadoUsuarioForm form) {
        if (form.activo == null) {
            return ResponseEntity.badRequest().build();
        }
        try {
            Docente docente = service.actualizarEstado(id, form.activo);
            return ResponseEntity.ok(docente);
        } catch (RuntimeException e) {
            return ResponseEntity.notFound().build();
        }
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void eliminar(@PathVariable Integer id) {
        service.eliminar(id);

    }
}

File: Integraupt-Backend/BackendUsuarios/controllers/DocenteController.php
<?php

namespace Controllers;

use Config\App;
use Services\Docente\DocenteService;

class DocenteController
{
    private DocenteService $service;

    public function __construct()
    {
        $this->service = new DocenteService();
    }

    /** GET /api/docentes */
    public function listar(): void
    {
        App::json($this->service->listar());
    }

    /** GET /api/docentes/{id} */
    public function obtener(int $id): void
    {
        try {
            App::json($this->service->obtener($id));
        } catch (\RuntimeException $e) {
            App::notFound($e->getMessage());
        }
    }

    /** POST /api/docentes */
    public function crear(): void
    {
        try {
            $body = App::getBody();
            App::json($this->service->registrar($body), 201);
        } catch (\RuntimeException $e) {
            App::error($e->getMessage());
        }
    }

    /** PUT /api/docentes/{id} */
    public function actualizar(int $id): void
    {
        try {
            $body = App::getBody();
            App::json($this->service->actualizar($id, $body));
        } catch (\RuntimeException $e) {
            App::error($e->getMessage());
        }
    }

    /** PATCH /api/docentes/{id}/estado */
    public function actualizarEstado(int $id): void
    {
        $body = App::getBody();
        if (!isset($body['activo'])) {
            App::error('El campo activo es requerido.');
        }
        try {
            App::json($this->service->actualizarEstado($id, (bool)$body['activo']));
        } catch (\RuntimeException $e) {
            App::notFound($e->getMessage());
        }
    }

    /** DELETE /api/docentes/{id} */
    public function eliminar(int $id): void
    {
        try {
            $this->service->eliminar($id);
            http_response_code(204);
        } catch (\RuntimeException $e) {
            App::notFound($e->getMessage());
        }
    }
}
